components/Search/SearchBar.tsx

```diff
@@ -0,0 +1,184 @@
+"use client";
+import React, { useState } from "react";
+import debounce from "lodash/debounce";
+import "./searchbar.css";
+import Link from "next/link";
+import { useOutsideClick } from "@/utils/hooks";
+
+interface Results {
+  posts: Post[];
+  tags: Tag[];
+  users: User[];
+}
+interface Post {
+  post_id: number;
+  title: string;
+  description: string;
+}
+interface Tag {
+  tag_id: number;
+  tag: string;
+}
+interface User {
```

Comment:
having many sources of truth re the user model doesn't seem right to me? can't you import from `@types/models`?